geometry/geometry_config.py

```diff
@@ -412,7 +412,7 @@
     if muShieldDesign > 6:  c.hadronAbsorber.length =     0*u.m # magnetized, counted inside muonshield 
     else:                   c.hadronAbsorber.length =  3.00*u.m
     c.hadronAbsorber.z     =  c.muShield.z - c.muShield.length/2. - c.hadronAbsorber.length/2.
-    if muShieldDesign > 9:  c.hadronAbsorber.WithConstField =  True
+    if muShieldDesign > 9 and muShieldDesign != 11:  c.hadronAbsorber.WithConstField =  True
```

Comment:
In order to minimize the number of configuation variables, muShieldDesign had been used as a variable to cover 2-3 cases. In the meantime, the number of use cases is increasing:
realistic field for hadron absorber: yes or no
realistic field for muon shield: yes or no
muon shield design itself: X

maybe instead of using a linear increasing number to cover everything, we could either 
1) change muon shield design itself: X + 100*(realistic field for hadron absorber) + 1000(realistic field for muon shield)
 2) promote hadronAbsorber.WithConstField and muShieldDesign.withConstMuonShieldField to be set in run_simScript.py

Reply:
This is a good idea in general, but I would suggest two things:

* We could use an options object with several different properties to pass this information to the muon shield. These magic numbers are very good when it comes to saving space in e.g. detector IDs, but in this case I don't there's a constraint in size etc., so it might be preferable to have an object which can be interrogated interactively and documented using the normal python facilities. This is AFAIK also a recommended way to do this in more modern C++ and would allow refactoring a few methods within the muon shield such that their argument lists aren't out of control anymore (the number of arguments has in some cases completely proliferated and I remember a significant bug we had a few years ago due to a default argument where it shouldn't have been).
* If we go along the numeric route, we should be very careful, as some checks in the muon shield currently are implemented as inequalities on the config, e.g. do something for muon shield configs >=7.

Reply:
Realistic field is strictly bounded to the certain shield design. 
So we can do this:
Replace the call of ShipMuonShield constructor to the one with geometry_config as input (solves some additional problems) + add the possibility to set "c.hadronAbsorber.WithConstField=False" as an option to the run_simScript (unlike fieldmap it can be used almost independently of shield design).

Reply:
Realistic field is bounded to certain design only for now. We need to expect that the situation might change in the future. So it is indeed better to set up an object with shield parameters - it will make code much more readable and less bug prone. 
@olantwin what do you think? Shall we just create a Python dict with muon shield specifications or json object what would be read during run runsimscript run? If so, what parameters do you think we should put inside this object?

Reply:
There is potentially one more important use case, which you have to consider in addition to running run_simScript.py. One should be able to re-create exactly the same configuration, geometry and field with the parameters stored in the geofile which is created when running run_simScript.py. At least until we have a database with a proper tags which provides this information.